lib/ansible/plugins/shell/__init__.py

```diff
@@ -163,7 +171,21 @@ def mkdtemp(self, basefile=None, system=False, mode=0o700, tmpdir=None):
 
         return cmd
 
-    def expand_user(self, user_home_path, username=''):
+    def _mkdtemp2(self, basefile=None, system=False, mode=0o700, tmpdir=None) -> _ShellCommand:
```

Comment:
Can you add type annotations for the args on all the new functions here and in `powershell.py`? Adding them on the function signatures you updated would be good too.

Reply:
Thanks for the pickup they have been added.